Queueomatic/Queueomatic.DataAccess/UnitOfWork/UnitOfWork.cs

```diff
@@ -1,19 +1,24 @@
 ﻿using Microsoft.EntityFrameworkCore;
 using Queueomatic.DataAccess.DataContexts;
+using Queueomatic.DataAccess.Repositories.Interfaces;
 
 namespace Queueomatic.DataAccess.UnitOfWork;
 
 public class UnitOfWork : IDisposable, IUnitOfWork
 {
     private bool _disposed;
     private readonly ApplicationContext _context;
-    //Add repositories below
 
-    public UnitOfWork(ApplicationContext context)
+    public UnitOfWork(ApplicationContext context, IUserRepository userRepository, IRoomRepository roomRepository)
     {
         _context = context;
+        UserRepository = userRepository;
+        RoomRepository = roomRepository;
     }
 
+    public IUserRepository UserRepository { get; }
+    public IRoomRepository RoomRepository { get; }
```

Comment:
This would directly contradict our core principles, that is SOLID, specifically the D - Dependency Inversion. So I have to disagree with that decision.

Reply:
I understand and what you are saying is right!

Reply:
In addition, if you feel like program.cs gets too cluttered, then we can simply create extension methods to reduce the number of lines displayed.

Reply:
Good suggestion.